# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/crosscall_params.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/crosscall_params.h
--- a/security/sandbox/chromium/sandbox/win/src/crosscall_params.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/crosscall_params.h	Wed Jun 12 11:10:48 2019 +0100
@@ -5,8 +5,14 @@
 #ifndef SANDBOX_SRC_CROSSCALL_PARAMS_H__
 #define SANDBOX_SRC_CROSSCALL_PARAMS_H__
 
+#if !defined(SANDBOX_FUZZ_TARGET)
 #include <windows.h>
+
 #include <lmaccess.h>
+#else
+#include "sandbox/win/fuzzer/fuzzer_types.h"
+#endif
+
 #include <stddef.h>
 #include <stdint.h>
 
@@ -83,7 +89,7 @@
   // of this value depends on the specific service.
   union {
     NTSTATUS nt_status;
-    DWORD    win32_result;
+    DWORD win32_result;
   };
   // Number of extended return values.
   uint32_t extended_count;
@@ -111,19 +117,15 @@
 
   // Returns the beggining of the buffer where the IPC params can be stored.
   // prior to an IPC call
-  const void* GetBuffer() const {
-    return this;
-  }
+  const void* GetBuffer() const { return this; }
 
   // Returns how many parameter this IPC call should have.
   uint32_t GetParamsCount() const { return params_count_; }
 
   // Returns a pointer to the CrossCallReturn structure.
-  CrossCallReturn* GetCallReturn() {
-    return &call_return;
-  }
+  CrossCallReturn* GetCallReturn() { return &call_return; }
 
-  // Returns TRUE if this call contains InOut parameters.
+  // Returns true if this call contains InOut parameters.
   bool IsInOut() const { return (1 == is_in_out_); }
 
   // Tells the CrossCall object if it contains InOut parameters.
@@ -237,14 +239,13 @@
       return false;
     }
 
-    char* dest = reinterpret_cast<char*>(this) +  param_info_[index].offset_;
+    char* dest = reinterpret_cast<char*>(this) + param_info_[index].offset_;
 
     // We might be touching user memory, this has to be done from inside a try
     // except.
     __try {
       memcpy(dest, parameter_address, size);
-    }
-    __except(EXCEPTION_EXECUTE_HANDLER) {
+    } __except (EXCEPTION_EXECUTE_HANDLER) {
       return false;
     }
 
@@ -253,8 +254,7 @@
     if (is_in_out)
       SetIsInOut(true);
 
-    param_info_[index + 1].offset_ = Align(param_info_[index].offset_ +
-                                                size);
+    param_info_[index + 1].offset_ = Align(param_info_[index].offset_ + size);
     param_info_[index].size_ = size;
     param_info_[index].type_ = type;
     return true;
@@ -270,12 +270,12 @@
   uint32_t GetSize() const { return param_info_[NUMBER_PARAMS].offset_; }
 
  protected:
-  ActualCallParams() : CrossCallParams(0, NUMBER_PARAMS) { }
+  ActualCallParams() : CrossCallParams(0, NUMBER_PARAMS) {}
 
  private:
   ParamInfo param_info_[NUMBER_PARAMS + 1];
-  char parameters_[BLOCK_SIZE - sizeof(CrossCallParams)
-                   - sizeof(ParamInfo) * (NUMBER_PARAMS + 1)];
+  char parameters_[BLOCK_SIZE - sizeof(CrossCallParams) -
+                   sizeof(ParamInfo) * (NUMBER_PARAMS + 1)];
   DISALLOW_COPY_AND_ASSIGN(ActualCallParams);
 
   friend uint32_t GetMinDeclaredActualCallParamsSize(uint32_t param_count);